R/transformAbundance.R

```diff
@@ -0,0 +1,415 @@
+                                                          transform = transform,
+                                                          pseudocount = pseudocount,
+                                                          scalingFactor = scalingFactor)
+
+              # Assign transformed table to assays
+              assay(x, name) <- transformed_table
+
+              return(x)
+
+          }
+)
+
+##################################Z-TRANSFORM###################################
+
+#' @rdname transformAbundance
+#' @export
+setGeneric("ZTransform", signature = c("x"),
+           function(x,
+                    abund_values = "counts",
+                    name = "ZTransform",
+                    pseudocount = FALSE,
+                    scalingFactor = 1)
+               standardGeneric("ZTransform"))
+
+
+#' @rdname transformAbundance
+#' @export
+setMethod("ZTransform", signature = c(x = "SummarizedExperiment"),
+          function(x,
+                   abund_values = "counts",
+                   name = "ZTransform",
+                   pseudocount = FALSE,
+                   scalingFactor = 1){
+
+              # Input check
+              # Check abund_values
+              .check_abund_values(abund_values, x)
+
+              # Check name
+              if(!.is_non_empty_string(name)){
+                  stop("'name' must be a non-empty single character value.",
+                       call. = FALSE)
+              }
+
+              # Check pseudocount
+              if(!(pseudocount==FALSE || is.numeric(pseudocount))){
+                  stop("'pseudocount' must be FALSE or numeric value.",
+                       call. = FALSE)
+              }
+
+              # Check scalingFactor
+              if(!is.numeric(scalingFactor)){
+                  stop("'scalingFactor' must be a numeric value.",
+                       call. = FALSE)
+              }
+
+              # Get transformed table
+              transformed_table <- .get_ztransformed_table(assay = assay(x, abund_values),
+                                                          pseudocount = pseudocount,
+                                                          scalingFactor = scalingFactor)
+
+              # Assign transformed table to assays
+              assay(x, name) <- transformed_table
+
+              return(x)
+          }
+)
+
+###########################HELP FUNCTIONS####################################
+
+
+# Chooses which transformation function is applied
+.get_transformed_table <- function(assay, transform, pseudocount, scalingFactor){
+
+    # If "pseudocount" is TRUE or over 0 or transform is log10p, add pseudocount
+    if(!pseudocount==FALSE || transform=="log10p"){
+        # In case of log10p, pseudocount can be still FALSE. Then 1 is added.
+        if(pseudocount == FALSE){
+            # Add 1 as a pseudo count
+            assay <- assay + 1
+            warning("Transform was calculated with pseudocount value 1")
```

Comment:
Imo this warning should include why a warning is thrown. `Transform was calculated with pseudocount value '1'. Set pseudocount to a numeric value to resolve this warning.`

Reply:
Warning changed